tsconfig.json

```diff
@@ -25,6 +25,7 @@
     "emitDecoratorMetadata": true,
     // TODO: Type errors
     "useUnknownInCatchVariables": false,
+    "target": "es2021"
```

Comment:
This *shouldn't* be required if we're extending from tsconfig?

Reply:
I think we just need to update Typescript

Reply:
> This shouldn't be required if we're extending from tsconfig?

True, but this is here to override the `"target"` inherited from `@tsconfig/node18/tsconfig.json`, which is `"es2022"`.  For some reason, using that value breaks a few of our CI checks: `config`, `metrics-docs`, and `test (18/20)`.  They fail when run locally, too.  Here's an example of a failed CI run:  https://github.com/matrix-org/matrix-hookshot/actions/runs/4811143763

Tweaking `package.json` to either revert any updates made in this PR, or update some other deps to their latest versions, doesn't help.

It also works to extend `@tsconfig/node16/tsconfig.json`, which sets `"target": "es2021"`, along with some other minor differences from `node18`.  It turns out the only breaking difference is `"target"`, so I figured it was a more future-facing change to to override `"target"` to its `node16` default and let the rest be inherited from `node18`.

> I think we just need to update Typescript

It was already updated to its latest version in 5632112b, which at the time of writing is still the latest version (^5.0.4).

Reply:
Solved in 4d87454

Reply:
Not solved, reverted, and #729 created.